packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -183,87 +183,122 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
       final Offset position = switch (textDirection) {
-        TextDirection.rtl => Offset(size.height / 2, size.height / 2),
-        TextDirection.ltr => Offset(size.width - size.height / 2, size.height / 2),
+        TextDirection.rtl => Offset(maxRadius, maxRadius),
+        TextDirection.ltr => Offset(size.width - maxRadius, maxRadius),
       };
       canvas.drawCircle(position, radius, indicatorPaint);
     }
 
-    // Draw the stop indicator.
-    if (value != null && stopIndicatorRadius != null && stopIndicatorRadius! > 0) {
-      drawStopIndicator();
-    }
+    final double trackGapFraction = effectiveTrackGap / size.width;
+    final double? effectiveValue = value == null ? null : clampDouble(value!, 0.0, 1.0);
 
-    void drawActiveIndicator(double x, double width) {
-      if (width <= 0.0) {
-        return;
+    if (effectiveValue != null) {
+      final double trackStartFraction = trackGapFraction > 0
+          ? effectiveValue + math.min(effectiveValue, trackGapFraction)
+          : 0;
+
+      // Draw the track.
+      if (trackStartFraction <= 1) {
+        drawLinearIndicator(startFraction: trackStartFraction, endFraction: 1, color: trackColor);
       }
-      final Paint activeIndicatorPaint = Paint()..color = valueColor;
-      final double left = switch (textDirection) {
-        TextDirection.rtl => size.width - width - x,
-        TextDirection.ltr => x,
-      };
 
-      final Rect activeRect = Offset(left, 0.0) & Size(width, size.height);
-      if (indicatorBorderRadius != null) {
-        final RRect activeRRect = indicatorBorderRadius!.resolve(textDirection).toRRect(activeRect);
-        canvas.drawRRect(activeRRect, activeIndicatorPaint);
-      } else {
-        canvas.drawRect(activeRect, activeIndicatorPaint);
+      // Draw the stop indicator.
+      if (stopIndicatorRadius != null && stopIndicatorRadius! > 0) {
+        drawStopIndicator();
       }
+
+      // Draw the active indicator.
+      if (effectiveValue > 0) {
+        drawLinearIndicator(startFraction: 0, endFraction: effectiveValue, color: valueColor);
+      }
+
+      return;
     }
 
-    // Draw the active indicator.
-    if (value != null) {
-      drawActiveIndicator(0.0, clampDouble(value!, 0.0, 1.0) * size.width);
-    } else {
-      final double x1 = size.width * line1Tail.transform(animationValue);
-      final double width1 = size.width * line1Head.transform(animationValue) - x1;
+    final double firstLineHead = line1Head.transform(animationValue);
+    final double firstLineTail = line1Tail.transform(animationValue);
+    final double secondLineHead = line2Head.transform(animationValue);
+    final double secondLineTail = line2Tail.transform(animationValue);
+
+    // Draw the track before line 1.
+    if (firstLineHead < 1 - trackGapFraction) {
+      final double trackStartFraction = firstLineHead > 0
+          ? firstLineHead + math.min(firstLineHead, trackGapFraction)
```

Comment:
`math.min(firstLineHead, trackGapFraction)` is used in several places. I think it would be better if we create a method and call the method when we want to calculate the gap fraction, or, we add a comment for each use case of `math.min(...)`, just for better understanding and easier to maintain.

Something like:
```dart'
// This is to get the "lerped" track gap fraction so that when the currentValue is smaller than the `trackGapFraction`, the returned value is growing as the currentValue, up to `trackGapFraction`.
double calculateTrackGapFraction(double currentValue, double trackGapFraction ) {
    return math.min(currentValue, trackGapFraction);
}
```